reconciler/reconciler_test.go

```diff
@@ -301,9 +305,9 @@ func testReconciler(t *testing.T, batchOps bool) {
 	})
 
 	runTest("refreshing",
-		func(cfg *reconciler.Config[*testObject]) {
-			cfg.PruneInterval = 0
-			cfg.RefreshInterval = 500 * time.Millisecond
+		[]reconciler.Option{
+			reconciler.WithPruning(0), // Disable
```

Comment:
This feels like suboptimal DX. If pruning is on by default, we could alias this to `reconciler.WithoutPruning()` which reads easier.

Reply:
Agreed. Added `WithoutPruning`.